### PR TITLE
Add user as reactable

### PR DESCRIPTION
**Pull request by Zhao-Andy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
This allows the `User` model to be a `reactable`. This is part of prepping the backend for easier moderation, such as reporting with reacting to a spam acccount.

## Added to documentation?
- [x] no documentation needed

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -478,6 +478,10 @@ def receives_follower_email_notifications?
       email_follower_notifications
   end
 
+  def title
+    name
+  end
```
Comment (Zhao-Andy) [inline app/models/user.rb]: Technically don't need to do this in this PR. I think the `title` method is only for Algolia, but not entirely sure.

I ran into an issue where creating a `vomit` reaction on a user in console caused a 500 in `/internal/reports`, so proactively fixing this now.

File: app/models/reaction.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Reaction < ApplicationRecord
   counter_culture :user
 
   validates :category, inclusion: { in: CATEGORIES }
-  validates :reactable_type, inclusion: { in: %w[Comment Article] }
+  validates :reactable_type, inclusion: { in: %w[Comment Article User] }
```
Comment (rhymes) [inline app/models/reaction.rb]: how would you feel at moving `%w[Comment Article User]` to a constant? Kind of like we're doing in the previous line with `CATEGORIES`

File: app/models/user.rb (hunk 1 above)
Comment (rhymes) [reply app/models/user.rb]: makes sense!

File: app/models/reaction.rb (hunk 2 above)
Comment (Zhao-Andy) [reply app/models/reaction.rb]: 👍 makes sense!

**Pull request merged**